resources.storage.account.tf

```diff
@@ -170,23 +170,5 @@ resource "azurerm_storage_account" "storage" {
     }
   }
 
-  # Bug when nfsv3 is activated. The external resource azurerm_storage_account_network_rules is not taken into account
-  dynamic "network_rules" {
-    for_each = var.nfsv3_enabled ? ["enabled"] : []
-    content {
-      default_action             = "Deny"
-      bypass                     = var.network_bypass
-      ip_rules                   = local.storage_ip_rules
-      virtual_network_subnet_ids = var.subnet_ids
-      dynamic "private_link_access" {
-        for_each = var.private_link_access
-        content {
-          endpoint_resource_id = private_link_access.value.endpoint_resource_id
-          endpoint_tenant_id   = private_link_access.value.endpoint_tenant_id
-        }
-      }
-    }
-  }
-
-  tags = merge(local.default_tags, var.add_tags)
+  tags = merge({ "ResourceName" = format("%s", lower(replace(local.sa_name, "/[[:^alnum:]]/", "")), 0, 24) }, local.default_tags, var.add_tags)
```

Comment:
Function is blowing up when run.
```
╷
│ Error: Invalid function argument
│
│   on ../../resources.storage.account.tf line 173, in resource "azurerm_storage_account" "storage":
│  173:   tags = merge({ "ResourceName" = format("%s", lower(replace(local.sa_name, "/[[:^alnum:]]/", "")), 0, 24) }, local.default_tags, var.add_tags)
│     ├────────────────
│     │ while calling format(format, args...)
│
│ Invalid value for "args" parameter: too many arguments; only 1 used by format string.
```

I think this will work.
`format("%s", lower(substr(replace(local.sa_name, "/[[:^alnum:]]/", ""), 0, 24)))`

Reply:
took out lower(replace